src/topp/FalseDiscoveryRate.cpp

```diff
@@ -167,72 +138,38 @@ class TOPPFalseDiscoveryRate :
     // loading input
     //-------------------------------------------------------------
 
-    if (combined) // "in" was given
-    {
-      vector<PeptideIdentification> pep_ids;
-      vector<ProteinIdentification> prot_ids;
-
-      IdXMLFile().load(in, prot_ids, pep_ids);
+    vector<PeptideIdentification> pep_ids;
+    vector<ProteinIdentification> prot_ids;
 
-      try
-      {
-        if (!proteins_only)
-        {
-          fdr.apply(pep_ids);
-        }
-        if (!peptides_only)
-        {
-          fdr.apply(prot_ids);
-        }
-      }
-      catch (Exception::MissingInformation)
-      {
-        LOG_FATAL_ERROR << "FalseDiscoveryRate failed due to missing information (see above).\n";
-        return INCOMPATIBLE_INPUT_DATA;
-      }
+    IdXMLFile().load(in, prot_ids, pep_ids);
 
-      for (vector<ProteinIdentification>::iterator it = prot_ids.begin(); it != prot_ids.end(); ++it)
-      {
-        it->assignRanks();
-      }
-      for (vector<PeptideIdentification>::iterator it = pep_ids.begin(); it != pep_ids.end(); ++it)
-      {
-        it->assignRanks();
-      }
-
-      IdXMLFile().store(out, prot_ids, pep_ids);
-    }
-    else // "in_target" and "in_decoy" given
+    try
     {
-      vector<PeptideIdentification> pep_target, pep_decoy;
-      vector<ProteinIdentification> prot_target, prot_decoy;
-
-      IdXMLFile().load(in_target, prot_target, pep_target);
-      IdXMLFile().load(in_decoy, prot_decoy, pep_decoy);
-
-      //-------------------------------------------------------------
-      // calculations
-      //-------------------------------------------------------------
-
-      writeDebug_("Starting calculations with " + String(pep_target.size()) + " target and " + String(pep_decoy.size()) + " decoy peptide IDs", 1);
-
       if (!proteins_only)
       {
-        fdr.apply(pep_target, pep_decoy);
+        fdr.apply(pep_ids);
       }
       if (!peptides_only)
       {
-        fdr.apply(prot_target, prot_decoy);
+        fdr.apply(prot_ids);
       }
+    }
+    catch (Exception::MissingInformation)
+    {
+      LOG_FATAL_ERROR << "FalseDiscoveryRate failed due to missing information (see above).\n";
+      return INCOMPATIBLE_INPUT_DATA;
+    }
 
-      // TODO @all shouldn't ranks be assigned here as well?
-
-      //-------------------------------------------------------------
-      // writing output
-      //-------------------------------------------------------------
-      IdXMLFile().store(out, prot_target, pep_target);
+    for (vector<ProteinIdentification>::iterator it = prot_ids.begin(); it != prot_ids.end(); ++it)
```

Comment:
just a note: this bloat code is one of the reasons we should have a dedicated class for `vector<Pep|Prot>` things